### ISSUE TITLE
Service not updated on main service listing

### ISSUE THREAD

**Issue by stefwalter** [cockpit-project/cockpit#2591]
It seems sometimes a service is not updated in the main listing, as indicated by this test failure:

```
Wrote check-services-testBasic-FAIL.png
Journal database copied to check-services-testBasic-10.111.111.13-FAIL.journal
ln -s '/usr/lib/systemd/system/test.service' '/etc/systemd/system/default.target.wants/test.service'
EWarning: Permanently added '10.111.111.13' (ECDSA) to the list of known hosts.

======================================================================
ERROR: testBasic (__main__.TestServices)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-services", line 121, in testBasic
    wait_service_state("test-fail.service", "failed")
  File "./check-services", line 76, in wait_service_state
    b.wait_in_text(svc_sel(service), state)
  File "/home/hubbot/hubbot/172f5ef7ba9b825d653a2d67fcdbf2af64954d3f_r7_x86-64/test/testlib.py", line 250, in wait_in_text
    return self.wait_js_func('ph_in_text', selector, text)
  File "/home/hubbot/hubbot/172f5ef7ba9b825d653a2d67fcdbf2af64954d3f_r7_x86-64/test/testlib.py", line 223, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))), timeout=self.phantom_wait_timeout)
  File "/home/hubbot/hubbot/172f5ef7ba9b825d653a2d67fcdbf2af64954d3f_r7_x86-64/test/testlib.py", line 637, in wait
    return self.run({'cmd': 'wait', 'cond': cond, 'timeout': timeout*1000})
  File "/home/hubbot/hubbot/172f5ef7ba9b825d653a2d67fcdbf2af64954d3f_r7_x86-64/test/testlib.py", line 600, in run
    raise Error("timeout")
Error: timeout

----------------------------------------------------------------------
Ran 1 test in 105.900s

FAILED (errors=1)
```

In this case the service has failed, but when going back to the main display it still shows active:

*Failing Test Service test-fail.service active(running)*

**Comment by stefwalter** [comment]
![check-services-testbasic-fail](https://cloud.githubusercontent.com/assets/795070/9305581/b9787f8a-44f2-11e5-8292-5a601b84f13f.png)

**Comment by stefwalter** [comment]
The logs confirm that the service did indeed fail:

```
Aug 17 10:17:04 m13 cockpit-ws[2192]: localhost: remove channel 1:1!2 for session
Aug 17 10:17:04 m13 cockpit-ws[2192]: localhost: removed channel 1:1!2 for session
Aug 17 10:17:04 m13 cockpit-ws[2192]: 1: remove channel 1:1!2 for socket
Aug 17 10:17:04 m13 cockpit-ws[2192]: channel 1:1!2 does not exist
Aug 17 10:17:04 m13 cockpit-ws[2192]: localhost: added channel 1:1!3 to session
Aug 17 10:17:04 m13 cockpit-ws[2192]: 1: added channel 1:1!3 to socket
Aug 17 10:17:04 m13 cockpit-bridge[2253]: Failed to get realtime timestamp: Cannot assign requested address
Aug 17 10:17:04 m13 systemd[1]: Starting Failing Test Service...
Aug 17 10:17:04 m13 systemd[1]: Started Failing Test Service.
Aug 17 10:17:05 m13 systemd[1]: test-fail.service: main process exited, code=exited, status=1/FAILURE
Aug 17 10:17:05 m13 systemd[1]: Unit test-fail.service entered failed state.
Aug 17 10:17:05 m13 cockpit-ws[2192]: localhost: remove channel 1:1!3 for session
Aug 17 10:17:05 m13 cockpit-ws[2192]: localhost: removed channel 1:1!3 for session
Aug 17 10:17:05 m13 cockpit-ws[2192]: 1: remove channel 1:1!3 for socket
Aug 17 10:17:05 m13 cockpit-ws[2192]: dropping message with unknown channel 1:1!3 from session
Aug 17 10:17:05 m13 cockpit-ws[2192]: channel 1:1!3 does not exist
Aug 17 10:17:05 m13 sh[2318]: WORKING
Aug 17 10:17:10 m13 sh[2318]: WORKING
Aug 17 10:17:15 m13 sh[2318]: WORKING
Aug 17 10:17:20 m13 sh[2318]: WORKING
Aug 17 10:17:25 m13 sh[2318]: WORKING
Aug 17 10:17:30 m13 sh[2318]: WORKING
Aug 17 10:17:35 m13 sh[2318]: WORKING
Aug 17 10:17:40 m13 sh[2318]: WORKING
Aug 17 10:17:45 m13 sh[2318]: WORKING
Aug 17 10:17:50 m13 sh[2318]: WORKING
Aug 17 10:17:55 m13 sh[2318]: WORKING
Aug 17 10:18:00 m13 sh[2318]: WORKING
Aug 17 10:18:05 m13 sh[2318]: WORKING
Aug 17 10:18:07 m13 systemd[1]: Started OpenSSH Server Key Generation.
Aug 17 10:18:07 m13 systemd[1]: Starting OpenSSH per-connection server daemon...
Aug 17 10:18:07 m13 systemd[1]: Started OpenSSH per-connection server daemon.
Aug 17 10:18:07 m13 sshd[2350]: Accepted publickey for root from 10.111.111.200 port 55452 ssh2: RSA 9f:92:a1:54:9f:32:f7:40:17:39:0f:bc:4e
```

**Comment by mvollmer** [comment]
Maybe this was fixed by #2790...
